src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs

```diff
@@ -309,68 +362,124 @@ private ObjectPropertyInitializer[] GetPropertyInitializers(IReadOnlyList<Parame
+
         protected override IEnumerable<ObjectTypeProperty> BuildProperties()
-        {
-            foreach (var field in Fields)
-                yield return new ObjectTypeProperty(field, Fields.GetInputByField(field), this);
-        }
+            => Fields.Select(f => new ObjectTypeProperty(f, Fields.GetInputByField(f)));
 
         protected override IEnumerable<ObjectTypeConstructor> BuildConstructors()
         {
-            yield return InitializationConstructor;
-            if (SerializationConstructor != InitializationConstructor)
+            if (!_inputModel.IsUnknownDiscriminatorModel)
+            {
+                yield return InitializationConstructor;
+            }
+
+            if (SerializationConstructor != InitializationConstructor && (!Configuration.Generation1ConvenienceClient || IncludeDeserializer))
+            {
                 yield return SerializationConstructor;
+            }
         }
 
         protected override bool EnsureHasJsonSerialization()
         {
             if (IsPropertyBag)
                 return false;
-            return true;
+            return _inputModelSerialization.Json;
         }
 
-        protected override bool EnsureHasXmlSerialization()
-        {
-            return false;
-        }
+        protected override bool EnsureHasXmlSerialization() => _inputModelSerialization.Xml is not null;
 
-        protected override bool EnsureIncludeSerializer()
-        {
-            return _inputModel.Usage.HasFlag(InputModelTypeUsage.Input);
-        }
+        protected override bool EnsureIncludeSerializer() => _inputModelUsage.HasFlag(InputModelTypeUsage.Input) && !_inputModel.IsUnknownDiscriminatorModel;
 
-        protected override bool EnsureIncludeDeserializer()
-        {
-            return _inputModel.Usage.HasFlag(InputModelTypeUsage.Output);
-        }
+        protected override bool EnsureIncludeDeserializer() => _inputModelUsage.HasFlag(InputModelTypeUsage.Output);
 
         protected override JsonObjectSerialization EnsureJsonSerialization()
         {
             // Serialization uses field and property names that first need to verified for uniqueness
             // For that, FieldDeclaration instances must be written in the main partial class before JsonObjectSerialization is created for the serialization partial class
-            return new(Type, SerializationConstructorSignature.Parameters, CreatePropertySerializations().ToArray(), null, Discriminator, false);
+            var properties = SerializationBuilder.GetPropertySerializations(this, _typeFactory);
+            var additionalProperties = CreateAdditionalPropertySerialization();
+            return new(Type, SerializationConstructor.Signature.Parameters, properties, additionalProperties, Discriminator, IncludeConverter);
```

Comment:
should we just put this into the serializationBuilder as we did for the xml serialization?

Reply:
We need access to internal `_inputModel` property of the base models.